src/App.tsx

```diff
@@ -1,61 +1,134 @@
 import './App.scss';
+import { ChangeEvent, useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { UsersList } from './components/UsersList/UsersList';
+import { TodoList } from './components/TodoList';
+import { User } from './types/user';
+import { Todo } from './types/todo';
+
+function getUser(userId: number): User | undefined {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || undefined;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [selectedUser, setSelectedUser] = useState('0');
+  const [title, setTitle] = useState('');
+  const [isClickedUser, setIsClickedUser] = useState(false);
+  const [isEmptyTitle, setIsEmptyTitle] = useState(false);
+  const isSomeFieldEmpty = !title || selectedUser === '0';
+
+  // eslint-disable-next-line max-len
+  const pattern = 'абвгдеёжзийклмнопрстуфхцчшщъыьэюяabcdefghijklmnopqrstuvwxyz1234567890 ';
+
+  const handleSelectChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(event.target.value);
+  };
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const targetValueArr = event.target.value.split('');
+    const lastSymbol = targetValueArr[targetValueArr.length - 1];
+
+    if (pattern.includes(lastSymbol.toLowerCase())) {
```

Comment:
why do you check only the last symbol? + you don't need to create an array to get it. `targetValueArr[targetValueArr.length - 1]` - the same logic works with string

Reply:
please, do validation in another way